src/app/view/Secure/SetPinView.tsx

```diff
@@ -144,8 +147,14 @@ const SetPinViewSimple = ({
               </Tooltip>
             </Grid>
 
-            <Button onPress={handleSecondInputSubmit}>
-              <Typography color="primary" variant="button">
+            <Button
+              onPress={handleSecondInputSubmit}
+              disabled={secondInput.length !== 4}
+            >
+              <Typography
+                color={secondInput.length !== 4 ? 'disabled' : 'primary'}
```

Comment:
nit: `const isPasswordComplete = (password) => password.length === 4` to factorize and make it very clear